Student/Student/Submissions/SubmissionDetails/SubmissionDetailsPresenter.swift

```diff
@@ -17,119 +17,80 @@
+    func reload()
+    func embed()
+    var navigationItem: UINavigationItem { get }
+}
 
 class SubmissionDetailsPresenter {
-    typealias PresenterFactory = (Bool) -> PresenterUseCase
-
     let context: Context
     let assignmentID: String
     let userID: String
     let env: AppEnvironment
     weak var view: SubmissionDetailsViewProtocol?
 
-    let useCaseFactory: PresenterFactory
-
-    let submissionFrc: FetchedResultsController<Submission>
-    let assignmentFrc: FetchedResultsController<Assignment>
-    let courseFrc: FetchedResultsController<Course>
+    lazy var submissions: Store<GetSubmission> = {
+        let useCase = GetSubmission(context: context, assignmentID: assignmentID, userID: userID)
+        return env.subscribe(useCase) { [weak self] in
+            self?.view?.embed()
+            self?.view?.reload()
+        }
+    }()
 
-    var selectedAttempt: Int = 0
-    var contentSubmission: Submission?
+    lazy var assignment: Store<GetAssignment> = {
+        let useCase = GetAssignment(courseID: context.id, assignmentID: assignmentID)
+        return env.subscribe(useCase) { [weak self] in
+            self?.view?.reload()
+        }
+    }()
 
-    init(env: AppEnvironment = .shared, view: SubmissionDetailsViewProtocol, context: Context,
-         assignmentID: String, userID: String, useCaseFactory: PresenterFactory? = nil) {
+    init(env: AppEnvironment = .shared, view: SubmissionDetailsViewProtocol, context: Context, assignmentID: String, userID: String) {
         self.context = context
         self.assignmentID = assignmentID
         self.userID = userID
         self.env = env
         self.view = view
-        self.useCaseFactory = useCaseFactory ?? { (force: Bool) in
-            return SubmissionDetailsUseCase(context: context, assignmentID: assignmentID, userID: userID)
-        }
-        self.submissionFrc = env.subscribe(Submission.self, .forUserOnAssignment(assignmentID, userID))
-        self.assignmentFrc = env.subscribe(Assignment.self, .details(assignmentID))
-        self.courseFrc = env.subscribe(Course.self, .details(context.id)) // TODO: support group contexts
-
-        self.submissionFrc.delegate = self
-        self.assignmentFrc.delegate = self
-        self.courseFrc.delegate = self
     }
 
     func viewIsReady() {
-        loadDataFromServer()
-        loadData()
+        submissions.refresh(force: true)
+        assignment.refresh(force: true)
     }
 
-    func loadDataFromServer(force: Bool = false) {
-        let useCase = useCaseFactory(force)
-        env.queue.addOperationWithErrorHandling(useCase, sendErrorsTo: view)
+    func submissionFor(attempt: Int) -> Submission? {
+        return submissions.filter({ $0.attempt == attempt }).first
     }
 
-    func loadData() {
-        guard let assignment = loadAssignment() else { return }
-        if let course = loadCourse() {
-            view?.updateNavBar(subtitle: assignment.name, color: course.color)
```

Comment:
This doesn't appear to ever be called anymore. At some point the color needs to be set, right?

Reply:
Oh good catch. I will add this back in.